### PR TITLE
Removed exception handler from Bond class

### PR DESCRIPTION
**Pull request by vishwanathjadhav**
Removed exception handling, it's not required, raising unnecessary exceptions

### CODE REVIEW

File: insights/parsers/tests/test_bond.py [hunk 1]
```diff
@@ -168,3 +170,5 @@ def test_bond_class():
     bond_obj_2 = Bond(context_wrap(BONDINFO_MODE_6, CONTEXT_PATH))
     assert bond_obj_2.bond_mode == '1'
     assert bond_obj_2.active_slave is None
+
+    bond_obj_2 = Bond(context_wrap(BONDINFO_UNKNOWN_BOND_MODE, CONTEXT_PATH))
```
Comment (xiangce) [inline insights/parsers/tests/test_bond.py]: Better add a check for the `unknown` bond mode

File: insights/parsers/bond.py [hunk 2]
```diff
@@ -38,22 +38,22 @@
...
-    ['3', '2', '3']
+    ['3', '3', '2']
     >>> bond_info.xmit_hash_policy
-    layer2
-    >>> bond_info.active_slave
-    enp17s0f0
+    'layer2'
 """
+#
+#
```
Comment (xiangce) [inline insights/parsers/bond.py]: Remove this 2 lines?

File: insights/parsers/bond.py [hunk 3]
```diff
@@ -38,22 +38,22 @@
...
-    ['eno1', 'eno2']
+    ['eth1', 'eth2']
     >>> bond_info.aggregator_id
-    ['3', '2', '3']
+    ['3', '3', '2']
     >>> bond_info.xmit_hash_policy
-    layer2
-    >>> bond_info.active_slave
-    enp17s0f0
+    'layer2'
```
Comment (xiangce) [inline insights/parsers/bond.py]: Why did you remove the check of `bond_info.active_slave`? Better keep it here

File: insights/parsers/bond.py [hunk 4]
```diff
@@ -38,22 +38,22 @@
 ``/proc/net/bonding``.
 
 Examples:
-    >>> bond_info = shared[Bond]
+    >>> type(bond_info)
+    <class 'insights.parsers.bond.Bond'>
     >>> bond_info.bond_mode
-    '0'
+    '4'
```
Comment (xiangce) [inline insights/parsers/bond.py]: The test result of the examples better match with above typical output.

File: insights/parsers/bond.py [hunk 5]
```diff
@@ -38,22 +38,22 @@
...
     >>> bond_info.partner_mac_address
-    None
+    '00:00:00:00:00:00'
     >>> bond_info.slave_interface
-    ['eno1', 'eno2']
+    ['eth1', 'eth2']
```
Comment (xiangce) [inline insights/parsers/bond.py]: Same to above

File: insights/parsers/bond.py [hunk 6]
```diff
@@ -92,8 +92,6 @@ def parse_content(self, content):
                 self._bond_mode = raw_mode
                 if raw_mode in BOND_PREFIX_MAP:
                     self._bond_mode = BOND_PREFIX_MAP[raw_mode]
```
Comment (xiangce) [inline insights/parsers/bond.py]: `self._bond_mode = BOND_PREFIX_MAP.get(raw_mode, raw_mode)`

File: insights/parsers/bond.py (hunk 3 above)
Comment (vishwanathjadhav) [reply insights/parsers/bond.py]: @xiangce - Test data mode type is different (its not active-backup mode so no active_slaves)

File: insights/parsers/bond.py (hunk 4 above)
Comment (vishwanathjadhav) [reply insights/parsers/bond.py]: @xiangce - Bond mode is 4, it's not active-backup(mode-0)

File: insights/parsers/bond.py (hunk 5 above)
Comment (vishwanathjadhav) [reply insights/parsers/bond.py]: @xiangce - Input data is different, that's why the changes are appeared.

File: insights/parsers/bond.py (hunk 2 above)
Comment (vishwanathjadhav) [reply insights/parsers/bond.py]: @xiangce - Done

File: insights/parsers/tests/test_bond.py (hunk 1 above)
Comment (vishwanathjadhav) [reply insights/parsers/tests/test_bond.py]: @xiangce - Done

File: insights/parsers/bond.py (hunk 6 above)
Comment (vishwanathjadhav) [reply insights/parsers/bond.py]: @xiangce - Done

File: insights/parsers/bond.py [hunk 7]
```diff
@@ -89,11 +88,8 @@ def parse_content(self, content):
         for line in get_active_lines(content):
             if line.startswith("Bonding Mode: "):
                 raw_mode = line.split(":", 1)[1].strip()
-                self._bond_mode = raw_mode
```
[the remark this answers is not in the document]
Comment (vishwanathjadhav) [reply insights/parsers/bond.py]: @xiangce - yup, done that

File: insights/parsers/tests/test_bond.py [hunk 8]
```diff
@@ -168,3 +170,6 @@ def test_bond_class():
     bond_obj_2 = Bond(context_wrap(BONDINFO_MODE_6, CONTEXT_PATH))
     assert bond_obj_2.bond_mode == '1'
     assert bond_obj_2.active_slave is None
+
+    bond_obj_3 = Bond(context_wrap(BONDINFO_UNKNOWN_BOND_MODE, CONTEXT_PATH))
+    assert bond_obj_3.bond_mode is None
```
[the remark this answers is not in the document]
Comment (vishwanathjadhav) [reply insights/parsers/tests/test_bond.py]: @xiangce  - Added  one more mapping entry for `fault-tolerance (active-backup) (fail_over_mac active)` cause this is another notation for active-backup mode.

File: insights/parsers/bond.py (hunk 3 above)
Comment (vishwanathjadhav) [reply insights/parsers/bond.py]: Done

**Comment by xiangce** [comment]
@csams - Thanks, I agree with you. Will talk with @vishwanathjadhav to revert the logic.

**Comment by vishwanathjadhav** [comment]
@csams , @xiangce - Reverted the changes please check
